runners/jet/src/main/java/org/apache/beam/runners/jet/JetTransformTranslators.java

```diff
@@ -0,0 +1,443 @@
+import org.apache.beam.runners.jet.processors.ParDoP;
+import org.apache.beam.runners.jet.processors.StatefulParDoP;
+import org.apache.beam.runners.jet.processors.TestStreamP;
+import org.apache.beam.runners.jet.processors.ViewP;
+import org.apache.beam.runners.jet.processors.WindowGroupP;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.runners.TransformHierarchy.Node;
+import org.apache.beam.sdk.testing.TestStream;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.DoFnSchemaInformation;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.beam.sdk.values.PValue;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.WindowingStrategy;
+
+@SuppressWarnings("unchecked")
+class JetTransformTranslators {
+
+  /** A map from a Transform URN to the translator. */
+  private static final Map<String, JetTransformTranslator> TRANSLATORS = new HashMap<>();
+
+  static {
+    TRANSLATORS.put(PTransformTranslation.READ_TRANSFORM_URN, new ReadSourceTranslator());
+    TRANSLATORS.put(PTransformTranslation.CREATE_VIEW_TRANSFORM_URN, new CreateViewTranslator());
+    TRANSLATORS.put(PTransformTranslation.PAR_DO_TRANSFORM_URN, new ParDoTranslator());
+    TRANSLATORS.put(PTransformTranslation.GROUP_BY_KEY_TRANSFORM_URN, new GroupByKeyTranslator());
+    TRANSLATORS.put(PTransformTranslation.FLATTEN_TRANSFORM_URN, new FlattenTranslator());
+    TRANSLATORS.put(PTransformTranslation.ASSIGN_WINDOWS_TRANSFORM_URN, new WindowTranslator());
+    TRANSLATORS.put(PTransformTranslation.IMPULSE_TRANSFORM_URN, new ImpulseTranslator());
+    TRANSLATORS.put(PTransformTranslation.TEST_STREAM_TRANSFORM_URN, new TestStreamTranslator());
+  }
+
+  static JetTransformTranslator<?> getTranslator(PTransform<?, ?> transform) {
+    String urn = PTransformTranslation.urnForTransformOrNull(transform);
+    return urn == null ? null : TRANSLATORS.get(urn);
+  }
+
+  private static class ReadSourceTranslator<T>
+      implements JetTransformTranslator<PTransform<PBegin, PCollection<T>>> {
+
+    @Override
+    public Vertex translate(Pipeline pipeline, Node node, JetTranslationContext context) {
+      AppliedPTransform<PBegin, PCollection<T>, PTransform<PBegin, PCollection<T>>>
+          appliedTransform =
+              (AppliedPTransform<PBegin, PCollection<T>, PTransform<PBegin, PCollection<T>>>)
+                  node.toAppliedPTransform(pipeline);
```

Comment:
You might want to change the caller to pass in the applied transform. Then you don't have to repeat this code for every transform translator.

Reply:
True, thank you, will do so in the next commit.